components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -669,12 +669,14 @@ void BravePrefProvider::UpdateCookieRules(ContentSettingsType content_type,
       const auto google_sign_in_rule =
           google_sign_in_content_setting_it->Next();
       // The embedding pattern for the cookie rule will be the primary pattern
-      // for the BRAVE_GOOGLE_SIGN_IN permission
-      // We want to get all subdomains for the cookie rule
-      const auto embedding_pattern =
-          ContentSettingsPattern::ToDomainWildcardPattern(
-              google_sign_in_rule.primary_pattern);
-
+      // for the BRAVE_GOOGLE_SIGN_IN permission.
+      // We want to get all subdomains for the cookie rule...
+      auto embedding_pattern = ContentSettingsPattern::ToDomainWildcardPattern(
+          google_sign_in_rule.primary_pattern);
+      // ... but if that doesn't work, fallback to stored pattern.
+      if (!embedding_pattern.IsValid()) {
```

Comment:
A test?

Reply:
I considered it, but could only reproduce for localhost.